index.js

```diff
@@ -15,6 +15,7 @@ function getStream(inputStream, opts) {
 
 	var p = new Promise(function (resolve, reject) {
 		stream = bufferStream(opts);
+		inputStream.on('error', reject);
```

Comment:
Also, probably better to just listen on `stream`. I believe the event propagates.